pawno/include/PPC_PlayerCommands.inc

```diff
@@ -2785,31 +2833,58 @@ COMMAND:convoymembers(playerid, params[])
 COMMAND:stats(playerid, params[])
 {
 	// Setup local variables
-	new StatsMsg[1000], Name[24], TitleMsg[128];
+	new StatsMsg[1000], Name[24], TitleMsg[128], OtherPlayer;
 
 	// Send the command to all admins so they can see it
 	SendAdminText(playerid, "/stats", params);
 
 	// Check if the player has logged in
 	if (APlayerData[playerid][LoggedIn] == true)
 	{
-		// Construct the stats
-		format(StatsMsg, 1000, "Kilometers driven: {00FF00}%f{FFFFFF}\n", (APlayerData[playerid][StatsMetersDriven] / 1000));
-		format(StatsMsg, 1000, "%sFinished trucker jobs: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsTruckerJobs]);
-		format(StatsMsg, 1000, "%sFinished convoy jobs: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsConvoyJobs]);
-		format(StatsMsg, 1000, "%sFinished busdriver routes: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsBusDriverJobs]);
-		format(StatsMsg, 1000, "%sFinished pilot jobs: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsPilotJobs]);
-		format(StatsMsg, 1000, "%sFinished courier jobs: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsCourierJobs]);
-		format(StatsMsg, 1000, "%sFinished roadworker jobs: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsRoadworkerJobs]);
-		format(StatsMsg, 1000, "%sFixed vehicles for assistance: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsAssistance]);
-		format(StatsMsg, 1000, "%sFinished mafia jobs: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsMafiaJobs]);
-		format(StatsMsg, 1000, "%sDelivered stolen mafia-loads: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsMafiaStolen]);
-		format(StatsMsg, 1000, "%sFined players: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsPoliceFined]);
-		format(StatsMsg, 1000, "%sJailed players: {00FF00}%i{FFFFFF}\n", StatsMsg, APlayerData[playerid][StatsPoliceJailed]);
-		// Get the player's name
-		GetPlayerName(playerid, Name, sizeof(Name));
-		// Construct the title for the dialog
-		format(TitleMsg, 128, "Statistics for %s:", Name);
+		if(sscanf(params, "U", OtherPlayer)) SendClientMessage(playerid, 0xFF0000AA, "Usage: \"/stats <playerid (optional)>\"");
+		else
+		{
+			if(OtherPlayer != INVALID_PLAYER_ID)
```

Comment:
There is something wrong.

- `/stats` does not display any statistics.
- `/stats 5` (id not online) shows my statistics as this is returning `-1` which `INVALID_PLAYER_ID` is the variable for.
- I could not test if it works on players who are actually online yet.

Maybe we should check first if an ID has been given and then we must check if that ID is correct (e.g. player online) by using  `IsPlayerConnected`. After that we can assign either the ID of the player who used the command or the other player id to a (new) variable, let us say `StatsPlayer` which can be used for the format lines to avoid duplication.